pkg/iam/policy/action.go

```diff
@@ -301,13 +296,31 @@ var actionConditionKeyMap = map[Action]condition.KeySet{
 			condition.S3XAmzServerSideEncryptionCustomerAlgorithm,
 			condition.S3XAmzMetadataDirective,
 			condition.S3XAmzStorageClass,
+			condition.S3ObjectLockRetainUntilDate,
+			condition.S3ObjectLockMode,
+			condition.S3ObjectLockLegalHold,
+		}, condition.CommonKeys...)...),
+
+	PutObjectRetentionAction: condition.NewKeySet(
+		append([]condition.Key{
+			condition.S3ObjectLockRemainingRetentionDays,
+			condition.S3ObjectLockRetainUntilDate,
+			condition.S3ObjectLockMode,
+		}, condition.CommonKeys...)...),
```

Comment:
s3:object-lock-legal-hold - can be a condition for PutObjectRetention as per https://docs.aws.amazon.com/AmazonS3/latest/dev/amazon-s3-policy-keys.html

Reply:
done